crates/task-impls/src/transactions.rs

```diff
@@ -165,20 +165,28 @@ impl<
                         .number_of_empty_blocks_proposed
                         .add(1);
 
+                    let validated_state = self.consensus.read().await.decided_state();
+
                     // Calculate the builder fee for the empty block
                     let Some(builder_fee) = null_block::builder_fee(
                         self.membership.total_nodes(),
+                        validated_state.as_ref(),
```

Comment:
not sure if we need to pass decided state here

Reply:
I suppose there's nothing explicitly wrong with this, what is your reservation?

Reply:
Spoke offline, this seems fine if it's the only source of the data. The consensus shared state is going to be hopefully phased out in the next sprint, so as long as this doesn't introduce additional lock contention (which I don't think it does), I don't have any qualms about this.